### ISSUE TITLE
Netdevice API

### ISSUE THREAD

**Issue by DBLouis** [rust-lang/libc#1909]
The "netdevice" API defined here: https://man7.org/linux/man-pages/man7/netdevice.7.html, seems to be missing.
I can do a PR with a little help on where to start.

**Comment by joshtriplett** [comment]
I don't think we should add things that have been obsoleted by netlink (for which I'd suggest the excellent [neli](https://github.com/jbaublitz/neli) crate), but to the extent there are ioctl interfaces that have no netlink equivalent, by all means.

**Comment by minghu6** [comment]
> I don't think we should add things that have been obsoleted by netlink (for which I'd suggest the excellent [neli](https://github.com/jbaublitz/neli) crate), but to the extent there are ioctl interfaces that have no netlink equivalent, by all means.

The `ioctl` doesn't be obsoleted by netlink actually in practice, See [this lwn article](https://lwn.net/Articles/897202/)

**Comment by xizheyin** [comment]
Is anyone doing it now? If not I can do this. But I may need some guidance as well, this is my first time writing code for this crate.

**Comment by tgross35** [comment]
It looks like most of the API mentioned on that manpage have since been added.
